### PR TITLE
Added error handling middleware

### PR DESCRIPTION
**Pull request by josipslavic**

### CODE REVIEW

**Review by anteljubic — APPROVED** [review]

File: src/controllers/user.controller.ts [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
-        role: user.role,
-        fullName: user.fullName,
-        alias: user.alias,
-      },
-      process.env.JWT_KEY as string,
-      {
-        expiresIn: process.env.JWT_EXPIRES_IN,
-      }
-    );
+      const token = jwt.sign(
```
Comment (JCukovic) [inline src/controllers/user.controller.ts]: extract this into separate method/function

File: src/controllers/user.controller.ts [hunk 2]
```diff
@@ -1,69 +1,73 @@
...
+      const existingUser = await this.userService.findUserByEmail(
+        req.body.email
+      );
+      if (!existingUser) throw commonErrors.invalidCredentials;
 
-    const isPasswordMatching = await compare(
-      req.body.password,
-      existingUser.password
-    );
+      const isPasswordMatching = await compare(
```
Comment (JCukovic) [inline src/controllers/user.controller.ts]: extract this as well

File: src/controllers/user.controller.ts [hunk 3]
```diff
@@ -1,69 +1,73 @@
...
-        role: existingUser.role,
-        fullName: existingUser.fullName,
-        alias: existingUser.alias,
-      },
-      process.env.JWT_KEY as string,
-      {
-        expiresIn: process.env.JWT_EXPIRES_IN,
-      }
-    );
+      const token = jwt.sign(
```
Comment (JCukovic) [inline src/controllers/user.controller.ts]: same as first comment, try to reuse logic where you can

**Review by JCukovic — CHANGES_REQUESTED** [review]

File: src/controllers/comment.controller.ts [hunk 4]
```diff
@@ -1,69 +1,76 @@
...
+      const existingComment = this.commentService.getCommentById(
+        req.params.commentId
+      );
+      if (!existingComment) throw commonErrors.commentNotFound;
 
-    await this.commentService.deleteComment(req.params.commentId);
+      await this.commentService.deleteComment(req.params.commentId);
 
-    return res.status(200).json({ message: DELETE_SUCCESS });
+      return res.status(statusCodes.ok).json({ message: DELETE_SUCCESS });
```
Comment (JCukovic) [inline src/controllers/comment.controller.ts]: extract success responses as well

File: src/middleware/errorHandler.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { NextFunction, Request, Response } from 'express';
+import { CustomError } from '../classes/CustomError';
+
+export function errorHandler(
+  err: CustomError,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  console.log('Middleware Error Hadnling');
```
Comment (JCukovic) [inline src/middleware/errorHandler.ts]: not sure if this console log is needed - if you think it is just correct the typo

**Review by JCukovic — CHANGES_REQUESTED** [review]

File: src/controllers/news.controller.ts [hunk 6]
```diff
@@ -1,159 +1,180 @@
...
-      return res.status(400).json({ errors: ['image must be provided'] });
-
-    // Check if breaking news already exists since only one is allowed to exist
-    if (req.body.isBreakingNews) {
-      const existingBreakingNews = await this.newsService.getBreakingNews();
-      if (existingBreakingNews)
-        return res
-          .status(400)
-          .json({ errors: [ERROR_MESSAGES.BREAKING_NEWS_LIMIT] });
+      return res.status(statusCodes.ok).json(news);
```
Comment (JCukovic) [inline src/controllers/news.controller.ts]: extract all responses not just the one i commented (try to reuse code where you can)

**Review by JCukovic — CHANGES_REQUESTED** [review]

**Review by JCukovic — APPROVED** [review]

**Pull request merged**